applications/vanilla/models/class.discussionmodel.php

```diff
@@ -1144,10 +1144,15 @@ public function calculate(&$discussion) {
             $discussion->Read = false;
             $discussion->CountUnreadComments = true;
         } else {
+            // If the category was marked explicitly read at some point, see if that applies here
             if ($category && !is_null($category['DateMarkedRead'])) {
-                // If the category was marked explicitly read at some point, see if that applies here
-                $discussion->DateLastViewed = self::maxDate($discussion->DateLastViewed, $category['DateMarkedRead']);
+                // If the discussion was created after the category was marked read and hasn't been viewed, reset DateLastViewed to null
```

Comment:
> If the discussion was created after the category was marked read [..]

How are we accounting for that here?

> [...] and hasn't been viewed [..]

If we're checking for the user's last-viewed date, wouldn't we use the discussion row's existing `DateLastViewed` property?

Reply:
Oops. That comment was for the old logic.